src/options.cpp

```diff
@@ -884,3 +885,21 @@ int zmq::options_t::getsockopt (int option_, void *optval_, size_t *optvallen_)
     errno = EINVAL;
     return -1;
 }
+
+bool zmq::options_t::is_valid (int option_) const
+{
+    bool valid = true;
+
+    if (connected) {
+        switch (option_) {
+        case ZMQ_SNDHWM:
```

Comment:
I just did SNDHWM and RCVHWM for now, presumably there are a load of other ones we could add.